src/components/plots/barPlot.ts

```diff
@@ -151,6 +166,114 @@ export module Plot {
     }
 
     /**
+     * Retrieves the closest PlotData to queryPoint.
+     *
+     * Bars containing the queryPoint are considered closest. If queryPoint lies outside
+     * of all bars, we return the closest in the dominant direction (x for horizontal
+     * charts, y for vertical) and break ties
+     *
+     * @param {Point} queryPoint The point to which dataset points should be compared
+     *
+     * @returns {PlotData} The PlotData closest to queryPoint
+     */
+    public getClosestPlotData(queryPoint: Point): PlotData {
+      var keys = this.datasetOrder();
+      var chartXExtent = { min: 0, max: this.width() };
+      var chartYExtent = { min: 0, max: this.height() };
+
+      var minDist = Infinity;
+      var closest: any;
+
+      var accessor: { [axis: string]: (p: Point) => number } = {
+        "x": (p) => p.x,
+        "y": (p) => p.y
+      };
+
+      var dominantAxis: string = this._isVertical ? "x" : "y";
+      var secondaryAxis: string = this._isVertical ? "y" : "x";
```

Comment:
We've been consistent with a primaryAttr / secondaryAttr throughout the codebase.  Let's rename to such